ssl/quic/quic_impl.c

```diff
@@ -1565,10 +1568,18 @@ SSL *ossl_quic_conn_stream_new(SSL *s, uint64_t flags)
 int ossl_quic_get_error(const SSL *s, int i)
 {
     QCTX ctx;
+    int net_error;
 
     if (!expect_quic(s, &ctx))
         return 0;
 
+    quic_lock(ctx.qc);
+    net_error = ossl_quic_channel_net_error(ctx.qc->ch);
+    quic_unlock(ctx.qc);
+
+    if (net_error)
+        return SSL_ERROR_SYSCALL;
+
     return ctx.is_stream ? ctx.xso->last_error : ctx.qc->last_error;
```

Comment:
If we are taking a lock we may as well load last_error while we hold the lock also.

Reply:
Good point!